### PR TITLE
DEV-995: use NODE_ENV instead of domain

### PR DESCRIPTION
**Pull request by harsha09**
#### What's this PR do?
As of now React is configured to tag hostname as environment in Sentry, this PR changes the environment tag to actual ENVIRONMENT from env variables like `{production, staging, test, dev, demo}`

#### How should this be manually tested?
Raising an exception / warning from an UI, sentry log will be generated with ENVIRONMENT `{production, staging, test, dev, demo}` as a tag instead of hostname.

#### Do any changes need to be made before deployment to staging? production? (adding environment variables, for example)?
Need to add an env variable
ENVIRONMENT `{production, staging, test, dev, demo}`

### CODE REVIEW

**Review by x110dc — CHANGES_REQUESTED** [review]

**Review by x110dc — APPROVED** [review]
LGTM.

File: spa/src/settings.js [hunk 1]
```diff
@@ -52,6 +52,9 @@ export const STRIPE_API_VERSION = resolveConstantFromEnv('STRIPE_API_VERSION', '
 export const STRIPE_CLIENT_ID = resolveConstantFromEnv('STRIPE_CLIENT_ID', 'test_1234');
 export const STRIPE_OAUTH_SCOPE = resolveConstantFromEnv('STRIPE_OAUTH_SCOPE', 'read_write');
 
+// Environment {production, staging, test, dev, demo}
+export const ENVIRONMENT = resolveConstantFromEnv('ENVIRONMENT');
```
Comment (benjaminEwhite) [inline spa/src/settings.js:56]: Should this also pass in a default value for second argument? That's a genuine question -- it's fine if the answer is no, I just saw that `resolveConstantFromEnv` can take a default value.

**Review by benjaminEwhite — APPROVED** [review]
This looks good to me.

File: spa/src/settings.js (hunk 1 above)
Comment (harsha09) [reply spa/src/settings.js:56]: I've removed the default value here as it get's the default value from Django's `ENVIRONMENT` env variable, which makes sense to me.

**Pull request merged**